src/mpid/ch3/src/ch3u_win_fns.c

```diff
@@ -340,12 +340,6 @@ int MPID_Win_set_info(MPIR_Win * win, MPIR_Info * info)
     /*************** check for info alloc_shm ***************/
     /********************************************************/
 
-    if (win->create_flavor == MPI_WIN_FLAVOR_CREATE)
-        win->info_args.alloc_shm = FALSE;
-    if (win->create_flavor == MPI_WIN_FLAVOR_ALLOCATE ||
-        win->create_flavor == MPI_WIN_FLAVOR_SHARED)
-        win->info_args.alloc_shm = TRUE;
-
```

Comment:
This code always resets alloc_shm at win_set_info, should be removed. The same for alloc_shared_noncontig in the following diff block.